### PR TITLE
feat: Added OS version to the OS details

### PR DESCRIPTION
**Pull request by RishivikramN**
## Changes
Added new regexs for matching with the OS versions from the user agent. Currently implemented the version details for the following OSes:
- Windows
- Mac
- IOS
- Android

If needed, I can add for the other OSes too.

fixes https://github.com/PostHog/posthog/issues/6268

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [x] Accounted for the impact of any changes across different browsers

### CODE REVIEW

File: src/utils.ts
Comment (benjackwhite) [inline src/utils.ts:1]: We already have an official `$os_version` property which we use in our mobile SDKs (see React Native for example) https://github.com/PostHog/posthog-js-lite/blob/32d74d2642168325a10cb1f9a6eab4f7a75ff0b8/posthog-react-native/src/types.ts#L41

Given that we don't want to break existing code paths based on the `$os` value, could we instead do this code as a separate check (or modify this function to return a tuple of `[$os, $os_version?]` ?

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: src/utils.ts
Comment (RishivikramN) [reply src/utils.ts:1]: yeah sure, will check it out 👍

Comment (RishivikramN) [reply src/utils.ts:1]: Pushed the mentioned changes. I hope I have understood the requirement correctly.

Comment (RishivikramN) [reply src/utils.ts:1]: No Problem, I will add it. If I have any doubts, I will ping you.

Comment (RishivikramN) [reply src/utils.ts:1]: I have added a unit test to test this function.

**Review by benjackwhite — APPROVED** [review]
Amazing! Thanks so much for bearing with it and getting the tests in as well. Great addition.

**Comment by RishivikramN** [comment]
> Amazing! Thanks so much for bearing with it and getting the tests in as well. Great addition.

Glad I can help

**Pull request merged**